internal/api/courses/users/list_test.go

```diff
@@ -0,0 +1,74 @@
+package users
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/model"
+	"github.com/edulinq/autograder/internal/util"
+)
+
+func TestList(test *testing.T) {
+	course := db.MustGetCourse(db.TEST_COURSE_ID)
+	usersMap, err := db.GetCourseUsers(course)
+	if err != nil {
+		test.Fatalf("Unable to get course users.")
+	}
+
+	users := make([]*model.CourseUser, 0, len(usersMap))
+	for _, user := range usersMap {
+		users = append(users, user)
+	}
+
+	testCases := []struct {
+		email     string
+		permError bool
+	}{
+		// Invalid permissions.
+		{"other@test.com", true},
+		{"student@test.com", true},
+
+		// Valid permissions.
+		{"admin@test.com", false},
+		{"owner@test.com", false},
```

Comment:
Maybe include grader in here? I'm not 100% sure but I'd assume they would have invalid permissions.